extension/httpforwarder/extension_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpforwarder
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/testutil"
+	"go.uber.org/zap"
+)
+
+type clientRequestArgs struct {
+	method  string
+	url     string
+	headers map[string]string
+	body    string
+}
+
+func TestExtension(t *testing.T) {
+	listenAt := testutil.GetAvailableLocalAddress(t)
+	tests := []struct {
+		name                        string
+		config                      *Config
+		expectedbackendStatusCode   int
+		expectedBackendResponseBody []byte
```

Comment:
likely want to ensure headers are preserved

Reply:
Done.